core/trino-main/src/main/java/io/trino/server/InternalCommunicationModule.java

```diff
@@ -27,19 +28,31 @@
     protected void setup(Binder binder)
     {
         InternalCommunicationConfig internalCommunicationConfig = buildConfigObject(InternalCommunicationConfig.class);
-        configBinder(binder).bindConfigGlobalDefaults(HttpClientConfig.class, config -> {
-            // Set defaults for all HttpClients in the same guice context
-            // so in case of any additions or alternations here an update in:
-            //   io.trino.server.security.jwt.JwtAuthenticatorSupportModule.JwkModule.configure
-            // and
-            //   io.trino.server.security.oauth2.OAuth2ServiceModule.setup
-            // may also be required.
-            config.setHttp2Enabled(internalCommunicationConfig.isHttp2Enabled());
-            config.setKeyStorePath(internalCommunicationConfig.getKeyStorePath());
-            config.setKeyStorePassword(internalCommunicationConfig.getKeyStorePassword());
-            config.setTrustStorePath(internalCommunicationConfig.getTrustStorePath());
-            config.setTrustStorePassword(internalCommunicationConfig.getTrustStorePassword());
-        });
+        if (internalCommunicationConfig.isHttpsRequired() && internalCommunicationConfig.getKeyStorePath() == null && internalCommunicationConfig.getTrustStorePath() == null) {
+            String sharedSecret = internalCommunicationConfig.getSharedSecret()
+                    .orElseThrow(() -> new IllegalArgumentException("Internal shared secret must be set when internal HTTPS is enabled"));
+            configBinder(binder).bindConfigDefaults(HttpServerConfig.class, config -> config.setAutomaticHttpsSharedSecret(sharedSecret));
+            configBinder(binder).bindConfigGlobalDefaults(HttpClientConfig.class, config -> {
+                config.setHttp2Enabled(internalCommunicationConfig.isHttp2Enabled());
+                config.setAutomaticHttpsSharedSecret(sharedSecret);
+                config.setNodeEnvironment("test");
+            });
+        }
+        else {
+            configBinder(binder).bindConfigGlobalDefaults(HttpClientConfig.class, config -> {
+                // Set defaults for all HttpClients in the same guice context
+                // so in case of any additions or alternations here an update in:
+                //   io.trino.server.security.jwt.JwtAuthenticatorSupportModule.JwkModule.configure
```

Comment:
I think we need to call `setAutomaticHttpsSharedSecret(false)` in both of these mentioned modules, since the intention is for those clients to be used with external services. Having the internal cert probably doesn't hurt, but seems better not to.

This is fine to do unconditionally, as the only way to set that default is from this module (so we're just clearing the default).